### PR TITLE
Converted Search to TypeScript

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
* Converted code to TypeScript
* Updated related code

### Checklist

* [x] have tested my change
* [x] updated relevant documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1270,7 +1270,7 @@
 				{
 					"id": "searchView",
 					"name": "Results",
-					"when": "code-for-ibmi:searchViewVisable == true"
+					"when": "code-for-ibmi:searchViewVisible == true"
```
Comment (worksofliam) [inline package.json:1273]: Blimey, I'm stupid. Nice spot

File: src/api/Search.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          })
+        }
+      }
+    }
+
+    return results;
+  }
+}
+
+function sanitizeSearchTerm(searchTerm: string): string {
```
Comment (worksofliam) [inline src/api/Search.ts:119]: Lovely touch.

**Review by worksofliam — APPROVED** [review]
Another excellent cleanup. Also.. thanks for catching my awful spelling :)

Also testing.. working as expect. Forgot how much I love the search UI.. I rarely use it.

**Comment by sebjulliand** [comment]
You're very welcome! 😁
It is a great feature indeed, making clever use of grep.
Now that I get a better understanding of how the search works, I'll eventually spend some brain juice on #890.

**Pull request merged**